pkg/synchronizer/synchronizer.go

```diff
@@ -105,7 +115,11 @@ func (in *Synchronizer) Sync(ctx context.Context, msg google.PubSubMessage) erro
 }
 
 func (in *Synchronizer) skipNonOwnedSecrets(ctx context.Context, msg google.PubSubMessage) error {
-	secret, err := in.clientset.CoreV1().Secrets(msg.GetNamespace()).Get(ctx, msg.GetSecretName(), metav1.GetOptions{})
+	namespace, err := in.getNamespaceFromProjectID(ctx, msg.GetProjectID())
```

Comment:
`msg.GetProjectID()` is misleading as it actually returns the project _number_ (which is the only reference to the project in the pubsub messages), and not the human-readable project IDs we've set. Should probably rename the references here to avoid future confusion.

Because of the above, this will probably not find a matching namespace. We could work around this by using the project number with the  `ResourceManager` to fetch the actual project ID for use here.

LGTM otherwise. Much more robust solution 👍

Reply:
aha, good catch! will implement